packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/KeyPathFlowable.kt

```diff
@@ -21,7 +21,17 @@ import kotlinx.coroutines.flow.Flow
 /**
  * A __flowable__ is an internal entity that supports listening to changes on the type [T] as a
  * [Flow].
+ *
+ * This comes in two variants: One variant for flows on things like Objects, Results and Lists that
+ * allow you to pass in keypaths to restrict notifications to certain properties and one variant
+ * for flows on entities that doesn't support this, e.g. flows on query aggregates or the Realm as
+ * whole.
  */
+
+internal interface KeyPathFlowable<T> {
+    fun asFlow(keyPaths: List<String>? = null): Flow<T>
```

Comment:
Is there any reason for using `List` and not `vararg`?

Reply:
This is the internal interface that just mirrors the public API which uses `List`. I was considering whether to use `vararg` or `List` in the public API but ended up with List as it makes it easier to extend the API later. I.e. there is the potential that projections will end up adding arguments to the `asFlow` method as well.